ui/src/app/pages/data/data-page.spec.tsx

```diff
@@ -45,102 +49,85 @@ describe('DataPage', () => {
   });
 
   const component = () => {
-    return mount(
-      <MemoryRouter>
-        <DataComponent hideSpinner={() => {}} showSpinner={() => {}} />
-      </MemoryRouter>
+    return renderWithRouter(
+      <DataComponent hideSpinner={() => {}} showSpinner={() => {}} />
     );
   };
 
   it('should render', async () => {
-    const wrapper = component();
-    await waitOneTickAndUpdate(wrapper);
-    await waitOneTickAndUpdate(wrapper);
-    expect(wrapper.exists()).toBeTruthy();
+    component();
+    await screen.findByLabelText('Please Wait');
```

Comment:
this is a common pattern as well as its opposite.  Should we add something like `expectSpinner()` and `waitForNoSpinner()` to react-test-helpers ?

Reply:
Good call. I added the functions and replaced all usages for the files modified in this PR.